test/TestMainframeStake.js

```diff
@@ -108,16 +100,160 @@ contract('MainframeStake', (accounts) => {
     assert.equal(updatedRequiredStake, 10, 'required stake not updated as expected')
   })
 
-  it('should successfully destroy itself', async () => {
-    await escrowContract.destroy()
-  })
-
   it('should successfully drain mistakenly sent tokens', async () => {
-    await tokenContract.transfer(stakeContract.address, 100, { from: accounts[0] })
+    await tokenContract.transfer(stakeContract.address, 200, { from: accounts[0] })
     const totalBalanceBefore = await tokenContract.balanceOf(stakeContract.address)
-    assert.equal(totalBalanceBefore, 100)
+    assert.equal(totalBalanceBefore, 200)
     await stakeContract.emergencyERC20Drain(tokenContract.address, { from: accounts[0], value: 0 })
     const totalBalanceAfter = await tokenContract.balanceOf(stakeContract.address)
     assert.equal(totalBalanceAfter, 0, 'incorrect total balance after draining tokens')
   })
+
+  it('should deposit successfully when staking', async () => {
+    const requiredStake = await stakeContract.requiredStake()
+    await tokenContract.approve(stakeContract.address, requiredStake, {from: accounts[0], value: 0})
+    await stakeContract.stake(accounts[0], {from: accounts[0], value: 0})
+    const depositorsBalance = await stakeContract.balanceOf(accounts[0])
```

Comment:
Rather than having multiple `await` one after the other, that makes the calls happen in series, it's faster to make unrelated calls in parallel using `Promise.all()`:

```js
const [depositorsBalance, totalDepositBalance, totalBalance] = await Promise.all([
  stakeContract.balanceOf(accounts[0]),
  stakeContract.totalDepositBalance(),
  tokenContract.balanceOf(stakeContract.address),
  utils.assertEvent(stakeContract, {event: 'Deposit'}),
])
```

Reply:
Ah that's strange, any idea what is causing the slowdown?
Anyways it's not an issue to leave them as-is, no need to spend much time on it.